File: CS/T328505/Form1.cs
using DevExpress.LookAndFeel;
using DevExpress.Skins;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Controls;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Globalization;
using System.Linq;
using System.Text;
using System.Threading;
using System.Windows.Forms;

namespace T328505 {
    public partial class Form1 : XtraForm {
        public Form1() {
            InitializeComponent();
        }
        private void Form1_Load(object sender, EventArgs e) {
            this.tokenEdit1.Properties.UseCustomFilter = true;
            this.tokenEdit1.Properties.CustomFilterTokens = new EventHandler<MyFilterEventArgs>(TokenEdit_CustomFilterText);           
        }
        private void TokenEdit_CustomFilterText(object sender, MyFilterEventArgs e) {
            if(this.checkEdit1.Checked)
                e.IsValidToken = e.Token.Description.Contains("Morais");
        }
    }
}

File: VB/T328505/Form1.vb
Imports DevExpress.LookAndFeel
Imports DevExpress.Skins
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.Controls
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Globalization
Imports System.Linq
Imports System.Text
Imports System.Threading
Imports System.Windows.Forms

Namespace T328505
    Partial Public Class Form1
        Inherits XtraForm

        Public Sub New()
            InitializeComponent()
        End Sub
        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
            Me.tokenEdit1.Properties.UseCustomFilter = True
            Me.tokenEdit1.Properties.CustomFilterTokens = New EventHandler(Of MyFilterEventArgs)(AddressOf TokenEdit_CustomFilterText)
        End Sub
        Private Sub TokenEdit_CustomFilterText(ByVal sender As Object, ByVal e As MyFilterEventArgs)
            If Me.checkEdit1.Checked Then
                e.IsValidToken = e.Token.Description.Contains("Morais")
            End If
        End Sub
    End Class
End Namespace
